app/controllers/bookings_controller.rb

```diff
@@ -61,14 +85,32 @@ def destroy
     end
   end
 
+  def get_city_wise_cleaner
+    @cleaner_temp = CitiesCleaner.where(city_id:City.find_by(name:params[:city]))
+    @cleaner = []
+    @cleaner_temp.each do |cl|
+      @cleaner << cl.cleaner
+    end
+    new
+    render :new
+  end
+
   private
     # Use callbacks to share common setup or constraints between actions.
     def set_booking
       @booking = Booking.find(params[:id])
     end
 
-    # Never trust parameters from the scary internet, only allow the white list through.
+    # select city for new or create
+    def select_city
+      @city = City.all
+    end
+
     def booking_params
-      params.require(:booking).permit(:customer_id, :cleaner_id, :date)
+      params.require(:booking).permit(:cleaner_id)
+    end
+
+    def check_date_conflict?(datetime,cleaner)
```

Comment:
@mayur-kambariya 

1. method name should be `is_date_conflict?`, usually in ruby boolean method starts with `is` and we should stick to it unless have a reason.
2. Query should only check the count. so use `.count` or `.exists` method of ActiveRecord.